draft-ietf-quic-http.md

```diff
@@ -1390,28 +1390,34 @@ for more details.
 
 An HTTP implementation MUST NOT send frames or requests which would be invalid
 based on its current understanding of the peer's settings.  All settings begin
-at an initial value, and are updated upon receipt of a SETTINGS frame.  For
-servers, the initial value of each client setting is the default value.
+at an initial value.  Each endpoint can use these values to send messages before
+the peer's SETTINGS frame has arrived.  This removes the need to wait for the
+SETTINGS frame before sending messages.  When the SETTINGS frame arrives, any
+settings are changed to their new values.
+
+For servers, the initial value of each client setting is the default value.
 
 For clients using a 1-RTT QUIC connection, the initial value of each server
-setting is the default value. When a 0-RTT QUIC connection is being used, the
+setting is the default value.  When a 0-RTT QUIC connection is being used, the
 initial value of each server setting is the value used in the previous session.
-Clients MUST store the settings the server provided in the session being resumed
-and MUST comply with stored settings until the current server settings are
-received.  A client can use these initial values to send requests before the
-server's SETTINGS frame has arrived.  This removes the need for a client to wait
-for the SETTINGS frame before sending requests.
+Clients SHOULD store the settings the server provided in the connection where
+resumption information was provided. A client MUST comply with either stored
+settings -- or default values, if no values are stored -- when attempting 0-RTT.
+Once a server has provided new settings, clients MUST comply with those values.
```

Comment:
This does smooth over the worst of this mess, but since the ordering issue is not obvious, I think it's worth calling out explicitly that the client may choose not to store the settings if things were received in the wrong order.